cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -401,9 +404,16 @@ readProjectLocalConfig verbosity projectRootDir = do
         projectPackages         = [ "./*.cabal" ],
 
         -- This is to automatically pick up deps that we unpack locally.
-        projectPackagesOptional = [ "./*/*.cabal" ]
+        projectPackagesOptional = [ "./*/*.cabal" ],
+
+        projectConfigProvenance = Set.singleton Implicit
       }
 
+    setExplicitProjectConfig :: ProjectConfig -> ProjectConfig
+    setExplicitProjectConfig config =
```

Comment:
Are you sure you don't want to make a helper function for this operation? The helper could be more robust: instead of obliterating the old provenance it would just add to it (unless obliterating it is what you want here? Hard to tell.)